buildSrc/src/main/java/org/elasticsearch/gradle/MavenFilteringHack.java

```diff
@@ -0,0 +1,43 @@
+ /*
+     * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+     * or more contributor license agreements. Licensed under the Elastic License
+     * 2.0 and the Server Side Public License, v 1; you may not use this file except
+     * in compliance with, at your election, the Elastic License 2.0 or the Server
+     * Side Public License, v 1.
+     */
+package org.elasticsearch.gradle;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.apache.tools.ant.filters.ReplaceTokens;
+import org.gradle.api.file.CopySpec;
+
+/**
+ * Gradle provides "expansion" functionality using groovy's SimpleTemplatingEngine (TODO: check name).
+ * However, it allows substitutions of the form {@code $foo} (no curlies). Rest tests provide
+ * some substitution from the test runner, which this form is used for.
+ *
+ * This class provides a helper to do maven filtering, where only the form {@code $\{foo\}} is supported.
+ *
+ * TODO: we should get rid of this hack, and make the rest tests use some other identifier
+ * for builtin vars
+ */
+public class MavenFilteringHack {
+    /**
+     * Adds a filter to the given copy spec that will substitute maven variables.
+     * 
+     */
+    static void filter(CopySpec copySpec, Map<String,String> substitutions) {
+        Map<String,String> mavenSubstitutions = new LinkedHashMap<>();
+        Map<String, Object> argMap = new LinkedHashMap<>();
+
+        substitutions.forEach((k,v) -> mavenSubstitutions.put("{" + k, v));
```

Comment:
v needs to be explicitly converted to a String here

Reply:
I've updated the code to do what the groovy code was doing, explicitly calling toString() on the incoming Map keys and values.  I've also updated the type arguments in the method signature, to indicate that we don't know what types of objects will be contained in the Map argument.